binsync/configuration.py

```diff
@@ -1,164 +1,94 @@
+        self.recent_projects = recent_projects
 
-    @classmethod
-    def correct_path(cls, binary_path):
-        # example config: /path/to/fauxware_files/.fauxware.bsconf
-        binary_path = pathlib.Path(binary_path)
-        config_name = pathlib.Path(f".{binary_path.name}.{BS_CONFIG_POSTFIX}")
-        config_dir = binary_path.parent
-        return str(config_dir.joinpath(config_name))
-
-
-class GlobalConfig(Config):
-    __slots__ = Config.__slots__ + (
-        "recent_bs_projects",
-        "ida_path",
-        "ghidra_path",
-        "angr_path",
-        "binja_path",
-        "gdb_path",
-    )
+    def save_project_data(self, binary_path, user=None, repo_path=None, remote=None):
+        project_data = {"binary_name": pathlib.Path(binary_path).name, "user": user, "repo_path": str(repo_path), "remote": remote}
+        projectData = ProjectData.get_from_state(project_data)
+        binary_hash = _hashfile(binary_path)
+        self.add_recent_project_data(binary_hash, projectData)
 
-    def __init__(self,
-                 path,
-                 recent_bs_projects=None,
-                 ida_path=None,
-                 ghidra_path=None,
-                 angr_path=None,
-                 binja_path=None,
-                 gdb_path=None,
-                 ):
-        super(GlobalConfig, self).__init__(GlobalConfig.correct_path(path))
+    def add_recent_project_data(self, binary_hash, projectData):
+        if self.recent_projects is None:
+            self.recent_projects = {}
 
-        self.recent_bs_projects = recent_bs_projects or []
-        self.angr_path = angr_path
-        self.ida_path = ida_path
-        self.ghidra_path = ghidra_path
-        self.binja_path = binja_path
-        self.gdb_path = gdb_path
+        if binary_hash not in self.recent_projects.keys():
+            self.recent_projects = _dict_insert(self.recent_projects, binary_hash, [])
 
-    @classmethod
-    def correct_path(cls, path):
-        if path is None:
-            path = os.getenv("HOME") or "~/"
-        path = pathlib.Path(path).expanduser().absolute()
-
-        if path.is_dir():
-            path = path.joinpath(BS_GLOBAL_CONFIG_FILENAME)
-        elif path.name != BS_GLOBAL_CONFIG_FILENAME:
-            l.warning("")
-            path = path.parent.joinpath(BS_GLOBAL_CONFIG_FILENAME)
-
-        return path
-
-    def add_recent_project_path(self, path, user):
-        if self.recent_bs_projects is None:
-            self.recent_bs_projects = []
-
-        self.recent_bs_projects.insert(0, f"{path}:{user}")
-        # only save the last 5 projects
-        self.recent_bs_projects = self.recent_bs_projects[0:5]
+        if {k: v for k, v in projectData.__getstate__().items() if v is not None} not in self.recent_projects[binary_hash]:
+            self.recent_projects[binary_hash].insert(0, projectData.__getstate__())
+
+        self.recent_projects[binary_hash] = self.recent_projects[binary_hash][0:5]
```

Comment:
Make the `[0:5]` more configurable. Like the `5` is clearly `max_recent_projects` or something like that.